src/services/authenticator/authenticatorService.ts

```diff
@@ -0,0 +1,97 @@
+import HTTPService from 'services/HTTPService';
+import { AuthEntity } from 'types/authenticator/auth_entity';
+import { Code } from 'types/authenticator/code';
+import ComlinkCryptoWorker from 'utils/comlink/ComlinkCryptoWorker';
+import { getEndpoint } from 'utils/common/apiUtil';
+import { getActualKey, getToken } from 'utils/common/key';
+import { logError } from 'utils/sentry';
+
+const ENDPOINT = getEndpoint();
+export const getAuthCodes = async (): Promise<Code[]> => {
+    const masterKey = await getActualKey();
+    try {
+        const authKeyData = await getAuthKey();
+        const cryptoWorker = await ComlinkCryptoWorker.getInstance();
+        const authentitorKey = await cryptoWorker.decryptB64(
+            authKeyData.encryptedKey,
+            authKeyData.header,
+            masterKey
+        );
+        // always fetch all data from server for now
+        const authEntity: AuthEntity[] = await getDiff(0);
+        const authCodes = await Promise.all(
+            authEntity
+                .filter((f) => !f.isDeleted)
+                .map(async (entity) => {
+                    const decryptedCode = await cryptoWorker.decryptMetadata(
+                        entity.encryptedData,
+                        entity.header,
+                        authentitorKey
+                    );
+                    try {
+                        return Code.fromRawData(entity.id, decryptedCode);
+                    } catch (e) {
+                        console.log('failed to parse code', e, entity.id);
```

Comment:
Let's not log `e`?

Reply:
Ack.
Btw, I don't think exception will contain any sensitive data from my limited testing of various cases. 
From my experience, most libs  throw error without the input params in the msg. Now, if we think about it, it's a good practice as that can result in accidental leaks during error logging.